### DISCUSSION TITLE
What's a subj parameter of a JetStream.PullSubscribe() method for?

### DISCUSSION THREAD

**Discussion by rkintzi** [nats-io/nats.go#1007] [Q&A] [answered]
What's a `subj` parameter of a `JetStream.PullSubscribe()` method for?

I observe following behavior:

1. When a consumer is created without the `FilterSubject` configuration option, then no matter what I pass as subject to the `js.PullSubscribe()`, the subscription fetches all the consumer's messages.
2. When a consumer is created with a non-empty `FilterSubject` option, `PullSubscribe` returns with an error, if the subject differs from that filter.

Is this parameter of any use?

**Comment by derekcollison** [comment, marked as answer]
If the call creates a pull consumer it will use the subject as a filter subject for the consumer. Consumers can filter messages from a parent stream.

If the consumer already exists it will select it based on the subject.

This behavior is being simplified in the next version of the JetStream API which is coming rsn.

**Reply by rkintzi** [reply]
Thanks for the explanation. It makes perfect sense. 

Where can I find some info/documentation about the next JstStream API version?

**Reply by derekcollison** [reply]
I believe we have finalized the design, mostly centered around consumers as you would imagine.

@tbeets may have some pointers to info.
